crates/router/src/connector/bluesnap.rs

```diff
@@ -152,6 +152,29 @@ impl ConnectorValidation for Bluesnap {
             ),
         }
     }
+
+    fn validate_psync_reference_id(
+        &self,
+        data: &types::PaymentsSyncRouterData,
+    ) -> Result<(), errors::ConnectorError> {
+        // if connector_transaction_id is present, psync can be made
+        if data
+            .request
+            .connector_transaction_id
+            .get_connector_transaction_id()
+            .is_ok()
+        {
+            return Ok(());
+        }
+        // if merchant_id is present, psync can be made along with attempt_id
+        let meta_data: CustomResult<bluesnap::BluesnapConnectorMetaData, errors::ConnectorError> =
+            connector_utils::to_connector_meta_from_secret(data.connector_meta_data.clone());
+
+        match meta_data {
+            Ok(_) => Ok(()),
+            Err(_) => Err(errors::ConnectorError::ResponseDeserializationFailed),
```

Comment:
Please maintain the error context, use
```suggestion
            error @ Err(_) => error.change_context(errors::ConnectorError::ResponseDeserializationFailed),
```

Reply:
I changed the return type to CustomResult to accommodate this